src/pages/Overview/OverviewPage.tsx

```diff
@@ -572,43 +548,33 @@ export class OverviewPage extends React.Component<OverviewProps, State> {
     // We are going to assume that if the user can create/update Istio AuthorizationPolicies in a namespace
     // then it can use the Istio Injection Actions.
     // RBAC allow more fine granularity but Kiali won't check that in detail.
-    let canWrite = false;
-    if (
-      this.state.permissions &&
-      this.state.permissions[nsInfo.name] &&
-      this.state.permissions[nsInfo.name][AUTHORIZATION_POLICIES]
-    ) {
-      const resourcePermission = this.state.permissions[nsInfo.name][AUTHORIZATION_POLICIES];
-      canWrite = resourcePermission.create && resourcePermission.update && resourcePermission.delete;
-    }
 
     if (serverConfig.istioNamespace !== nsInfo.name) {
       if (serverConfig.kialiFeatureFlags.istioInjectionAction && !serverConfig.kialiFeatureFlags.istioUpgradeAction) {
         namespaceActions.push({
           isGroup: false,
-          isSeparator: true,
-          isDisabled: false
+          isSeparator: true
         });
         const enableAction = {
           isGroup: false,
           isSeparator: false,
-          isDisabled: !canWrite,
           title: 'Enable Auto Injection',
-          action: (ns: string) => this.onAddRemoveAutoInjection(ns, true, false)
+          action: (ns: string) =>
+            this.setState({ showTrafficManagement: true, nsTarget: ns, opTarget: 'enable', kind: 'injection' })
```

Comment:
I understand this design but it's confusing for another developer.
If we want to concentrate some of the actions in the "TrafficManagement" component, I guess that component should get a better name as "OverviewNamespaceModal" or something in that way that would give hint to understand the logic.

From my perspective, now that component performs more tasks: annotate a namespace, create some Istio configuration for an entire namespace, and potentially more in the short term.